// congressional-district-finder/dist/github/unitedstates/getDistrictShape.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.DISTRICT_NOT_FOUND = exports.UNEXPECTED_DISTRICT_JSON = exports.endpoint = undefined;
exports.parseDistrictShape = parseDistrictShape;
exports.default = getDistrictShape;

var _requestPromiseNative = require('request-promise-native');

var _requestPromiseNative2 = _interopRequireDefault(_requestPromiseNative);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var endpoint = exports.endpoint = function endpoint(district) {
    return 'https://theunitedstates.io/districts/cds/2016/' + district + '/shape.geojson';
};

var UNEXPECTED_DISTRICT_JSON = exports.UNEXPECTED_DISTRICT_JSON = 'Unexpected JSON.';
var DISTRICT_NOT_FOUND = exports.DISTRICT_NOT_FOUND = function DISTRICT_NOT_FOUND(district) {
    return 'District: "' + district + '" not found.';
};

function parseDistrictShape(_ref) {
    var geometry = _ref.geometry,
        properties = _ref.properties;


    if (!geometry || !properties) {
        throw new Error(UNEXPECTED_DISTRICT_JSON);
    }

    var coordinates = geometry.coordinates;

    var districtCode = properties.Code;
    var name = properties.District;

    return {
        districtCode: districtCode,
        name: name,
        polygons: coordinates
    };
}

function getDistrictShape(district) {

    var options = {
        uri: endpoint(district),
        json: true
    };

    return _requestPromiseNative2.default.get(options).catch(function (err) {
        var statusCode = err.statusCode;

        if (statusCode === 404) {
            // eslint-disable-next-line babel/new-cap
            var error = new Error(DISTRICT_NOT_FOUND(district));
            error.statusCode = 404;
            throw error;
        }
        throw err;
    }).then(parseDistrictShape);
}
